### PR TITLE
Fixes for Compile errors related to macOS16 (Ventura)

### PR DESCRIPTION
**Pull request by moudgils**
## What does this PR do?

Compile fixes for Ventura upgrade.

## How was this PR tested?

Tested ASV

### CODE REVIEW

File: Gems/Atom/RHI/Metal/Code/Source/RHI/PipelineState.cpp [hunk 1]
```diff
@@ -168,7 +168,12 @@ namespace AZ
                 [depthStencilDesc release];
                 depthStencilDesc = nil;
             }
+#if defined(__IPHONE_16_0) || defined(__MAC_13_0)
```
Comment (AMZN-Olex) [inline Gems/Atom/RHI/Metal/Code/Source/RHI/PipelineState.cpp:171]: Is this okay by PAL standards?

Comment (moudgils) [reply Gems/Atom/RHI/Metal/Code/Source/RHI/PipelineState.cpp:171]: I think so. Even though the define macro uses platform names it is not a platform related update but an OS related update. We dont do PAL-ification for OS updates. So in this case we are protecting the api for specific OS version.

Comment (VickyAtAZ) [reply Gems/Atom/RHI/Metal/Code/Source/RHI/PipelineState.cpp:171]: How about the compatibility with the versions in the future? 
The new rasterSampleCount could be used for iphone 16 or any versions moving forward. 
The sampleCount should be versions which are lower then iphone 16 or  mac 13

**Review by VickyAtAZ — COMMENTED** [review]
One small issue

File: Gems/Atom/RHI/Metal/Code/Source/RHI/PipelineState.cpp (hunk 1 above)
Comment (lumberyard-employee-dm) [reply Gems/Atom/RHI/Metal/Code/Source/RHI/PipelineState.cpp:171]: Is this define for new versions of MacOS and iOS?

If MacOS 14 comes out is this still defined?

Shouldn't we check the  `__APPLE__` and `__IOS__` defines?

Comment (moudgils) [reply Gems/Atom/RHI/Metal/Code/Source/RHI/PipelineState.cpp:171]: So the two macros are guaranteed to always be defined going forward for every OS version but will not be defined for any versions before ios_16_0 and mac_13_0. That way rasterSampleCount will be picked up going forward but not backwards.

**Review by VickyAtAZ — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Gems/Atom/RHI/Metal/Code/Source/RHI/PipelineState.cpp (hunk 1 above)
Comment (moudgils) [reply Gems/Atom/RHI/Metal/Code/Source/RHI/PipelineState.cpp:171]: As an example here are details related to this from the header within macOS13.1
....
#define __MAC_11_6          110600
#define __MAC_12_0          120000
#define __MAC_12_1          120100
#define __MAC_12_2          120200
#define __MAC_12_3          120300
#define __MAC_13_0          130000
#define __MAC_13_1          130100
/* __MAC_NA is not defined to a value but is used as a token by macros to indicate that the API is unavailable */

**Pull request merged**